addons/point_of_sale/static/src/app/store/pos_store.js

```diff
@@ -1900,6 +1900,42 @@ export class PosStore extends Reactive {
             canCreateLots = true;
         }
 
+        const usedLots = this.models["pos.pack.operation.lot"].filter(
+            (lot) =>
+                lot.pos_order_line_id?.product_id?.id === product.id &&
+                lot.pos_order_line_id?.order_id?.state === "draft"
+        );
+
+        // Remove lot/serial names that are already used in draft orders
+        existingLots = existingLots.filter((lot) => {
+            const usedLotCount = usedLots.reduce(
+                (currentQty, lineLot) =>
+                    currentQty +
+                    ((lot.name === lineLot.lot_name && lineLot.pos_order_line_id?.qty) || 0),
+                0
+            );
+            return lot.product_qty > usedLotCount;
+        });
+
+        // Check if the input lot/serial name is already used in another order
+        const isLotNameUsed = (itemValue) => {
+            const currentOrderId = this.selectedOrder.id;
+            const totalQty = existingLots.find((lt) => lt.name == itemValue)?.product_qty || 0;
+            let usedQty = 0; // tracks used qty of that lot
+
+            const checklot = (lot) => {
+                usedQty += lot.pos_order_line_id?.qty || 0;
+                return usedQty >= totalQty;
+            };
+
+            return usedLots.some(
+                (lot) =>
+                    lot.pos_order_line_id?.order_id?.id != currentOrderId &&
+                    lot.lot_name === itemValue &&
+                    checklot(lot)
+            );
+        };
+
         const existingLotsName = existingLots.map((l) => l.name);
```

Comment:
```suggestion
        const existingLotsName = existingLots.filter((l) => !isLotNameUsed(l.name)).map((l) => l.name);
```

May I know why don't you filter the selection here? That way, you don't need to modify the `EditListPopup` component.

Reply:
@caburj 
- We are already filtering `existingLots` [here](https://github.com/odoo/odoo/pull/187176/files#diff-86ab5d4b3803e1a7fa3373b7225bc610cfdcfc70fb413673b71d52a752706137R1922), which ensures the dropdown list doesn't show the serial number from `usedLots`.

- However, users can still manually enter a serial number in the input field. The `isLotNameUsed` function is passed to the `EditListPopup` component to handle this. This function validates the entered serial number by checking if it exists in `usedLots`.